src/components/Editor/Runners/PyodideRunner/PyodideWorker.js

```diff
@@ -0,0 +1,286 @@
+  await vendoredPackage?.before();
+  if (vendoredPackage) {
+    return;
+  }
+
+  // If the import is for a module built into Python then do nothing.
+  let pythonModule;
+  try {
+    pythonModule = pyodide.pyimport(name);
+  } catch (_) {}
+  if (pythonModule) {
+    return;
+  }
+
+  // If the import is for a package built into Pyodide then load it.
+  // Built-ins: https://pyodide.org/en/stable/usage/packages-in-pyodide.html
+  await pyodide.loadPackage(name).catch(() => {});
+  let pyodidePackage;
+  try {
+    pyodidePackage = pyodide.pyimport(name);
+  } catch (_) {}
+  if (pyodidePackage) {
+    return;
+  }
+
+  // Ensure micropip is loaded which can fetch packages from PyPi.
+  // See: https://pyodide.org/en/stable/usage/loading-packages.html
+  if (!pyodide.micropip) {
+    await pyodide.loadPackage("micropip");
+    pyodide.micropip = pyodide.pyimport("micropip");
+  }
+
+  // If the import is for a PyPi package then load it.
+  // Otherwise, don't error now so that we get an error later from Python.
+  await pyodide.micropip.install(name).catch(() => {});
+};
+
+const vendoredPackages = {
+  turtle: {
+    before: async () => {
+      pyodide.registerJsModule("basthon", fakeBasthonPackage);
+      await pyodide.loadPackage(
+        `${process.env.PUBLIC_URL}/packages/turtle-0.0.1-py3-none-any.whl`,
+      );
+    },
+    after: () =>
+      pyodide.runPython(`
+      import turtle
+      import basthon
+
+      svg_dict = turtle.Screen().show_scene()
+      basthon.kernel.display_event({ "display_type": "turtle", "content": svg_dict })
+      turtle.restart()
+    `),
+  },
+  p5: {
+    before: async () => {
+      pyodide.registerJsModule("basthon", fakeBasthonPackage);
+      await pyodide.loadPackage([
+        "setuptools",
+        `${process.env.PUBLIC_URL}/packages/p5-0.0.1-py3-none-any.whl`,
+      ]);
+    },
+    after: () => {},
+  },
+  pygal: {
+    before: () => {
+      pyodide.registerJsModule("pygal", { ...pygal });
+      pygal.config.renderChart = (content) =>
+        postMessage({ method: "handleVisual", origin: "pygal", content });
+    },
+    after: () => {},
+  },
+  sense_hat: {
+    before: async () => {
+      pyodide.registerJsModule("_internal_sense_hat", {
+        ..._internal_sense_hat,
+      });
+      await pyodide.loadPackage([
+        "pillow",
+        `${process.env.PUBLIC_URL}/packages/sense_hat-0.0.1-py3-none-any.whl`,
+      ]);
+
+      _internal_sense_hat.config.pyodide = pyodide;
+      _internal_sense_hat.config.emit = (type) =>
+        postMessage({ method: "handleSenseHatEvent", type });
+    },
+    after: () => {
+      const {
+        pyodide,
+        emit,
+        sensestick,
+        start_motion_callback,
+        stop_motion_callback,
+        ...config
+      } = _internal_sense_hat.config;
+      postMessage({
+        method: "handleVisual",
+        origin: "sense_hat",
+        content: config,
+      });
+    },
+  },
+};
```

Comment:
🙌  this is really nice! Can be optimised for sure but its a really nice structure for expanding on packages 🥳